lib/hyperion/result_handling/result_maker.rb

```diff
@@ -35,6 +36,8 @@ def make_from_typho(typho_result, result_class)
       read_body = proc { read(typho_result.body, route.response_descriptor) }
       code = typho_result.code
 
+      log_error_response(read(typho_result.body, :json)) unless typho_result.success?
```

Comment:
The `ResultMaker` class is "private" to hyperion, so it is not tested directly; instead, it is tested through the public interface. Up until this point, `ResultMaker`was responsible only for unpacking a response according to the response content. [On success](https://github.com/indigobio/hyperion/blob/f-logs/lib/hyperion/result_handling/result_maker.rb#L41-L42), the body is the deserialized response body (tested [here](https://github.com/indigobio/hyperion/blob/f-logs/spec/lib/hyperion_spec.rb#L49-L53)). [On a 400-level error](https://github.com/indigobio/hyperion/blob/f-logs/lib/hyperion/result_handling/result_maker.rb#L53-L56), the body is a `ClientErrorResponse` (tested [here](https://github.com/indigobio/hyperion/blob/f-logs/spec/lib/hyperion_spec.rb#L60-L69)). Body is unspecified for other response statuses.

[The call to `log_error_response`](https://github.com/indigobio/hyperion/blob/f-logs/lib/hyperion/result_handling/result_maker.rb#L39) is passing an unspecified body to `read` when the response is a non-4xx failure. A better way to implement this is to always pass a fully formed `HyperionResult` to `log_error_response`. This has two advantages: 1) the logger gets to determine when to log (e.g., `unless result.success?`), and 2) the logger receives an object with well-specified structure (`HyperionResult`). Test the logging behavior in `logger_spec.rb`. If you feel it's appropriate, also verify that `Hyperion` calls `Logger#log_error_response` in `hyperion_spec.rb`.

(`ResultMaker` also does [a little work](https://github.com/indigobio/hyperion/blob/f-logs/lib/hyperion/result_handling/result_maker.rb#L21-L24) to enable the result-matching DSL, which is tested [here](https://github.com/indigobio/hyperion/blob/f-logs/spec/lib/hyperion_spec.rb#L88-L167).)